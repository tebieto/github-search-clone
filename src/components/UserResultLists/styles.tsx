import styled from 'styled-components';
import colors from '../../utils/colors';

export const UserResultListsContainer = styled.div`
  .user-result-list {
    background-color: ${colors.secondaryColor};
    display: inline-block;
    width: 700px;
    padding: 20px;
    border-radius: 5px;
    .title {
      display: flex;
      align-items: center;
      margin-bottom: 10px;
      .main-title {
        margin: 0;
        margin-right: 10px;
      }
      .sub-title {
        color: ${colors.tertiaryTextColor};
        font-size: 14px;
      }
    }
    .description {
      color: ${colors.tertiaryTextColor};
      font-size: 12px;
    }
  }
`;
