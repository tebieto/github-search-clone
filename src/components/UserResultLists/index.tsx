import React from 'react';
import { UserResultListsContainer } from './styles';
import Pagination from '../Pagination';
import { addCommaToNumber, customScrollToTop } from '../../utils';
interface UserNode {
  id: string;
  name: string;
  bio: string;
  email: string;
}
interface UserResultListsProps {
  user: UserNode[];
}
const UserResultLists = ({ user }: UserResultListsProps): JSX.Element => {
  const [currentPage, setCurrentPage] = React.useState(0);
  const handleSetCurrentPage = React.useCallback((page: number) => {
    setCurrentPage(page);
  }, []);

  React.useEffect(() => {
    customScrollToTop();
  }, [currentPage]);

  const maxItemPerPage = 10;

  return (
    <UserResultListsContainer>
      <h2>{addCommaToNumber(user.length)} User Results</h2>
      {user
        .slice(
          currentPage * maxItemPerPage,
          currentPage * maxItemPerPage + maxItemPerPage,
        )
        .filter(({ id }: UserNode) => id)
        .map((user: UserNode, key: number) => {
          return (
            <div key={key} className="user-result-list">
              <div className="title">
                <h4 className="main-title">{user.name || 'N/A'}</h4>
                <div className="sub-title">{user.email || 'N/A'}</div>
              </div>
              <div className="description">{user.bio || 'N/A'}</div>
            </div>
          );
        })}
      <Pagination
        itemsLength={user.length}
        currentPage={currentPage}
        setCurrentPage={handleSetCurrentPage}
        maxItemPerPage={maxItemPerPage}
      />
    </UserResultListsContainer>
  );
};

export default UserResultLists;
