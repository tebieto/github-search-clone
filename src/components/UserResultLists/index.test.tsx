import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserResultLists from '.';

it('renders correctly', () => {
  const tree = shallow(<UserResultLists user={[]} />);
  expect(toJson(tree)).toMatchSnapshot();
});
