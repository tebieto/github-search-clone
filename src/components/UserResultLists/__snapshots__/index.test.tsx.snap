// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <h2>
    0
     User Results
  </h2>
  <Pagination
    currentPage={0}
    itemsLength={0}
    maxItemPerPage={10}
    setCurrentPage={[Function]}
  />
</styled.div>
`;
